### DISCUSSION TITLE
Client does not support authentication protocol requested by server

### DISCUSSION THREAD

**Discussion by Alien0-7** [games647/FastLogin#1139] [Q&A] [answered]
hi i have a problem using MySQL
Console error: https://pastebin.com/uES69tN3
I also tried to use caching_sha2_password but it doesn’t work.
I saw that there is already written this error but they said that to solve it you have to downgrade the version of mysql but I can not do it in my server

**Comment by TuxCoding** [comment]
The issue is that the MariaDB (fork of MySQL) driver included in FastLogin doesn't support this new authentication `caching_sha2` and only supports the old default `mysql_native_password`.

>I saw that there is already written this error but they said that to solve it you have to downgrade the version of mysql but I can not do it in my server

You can change the default authentication system by editing the user properties using [this](https://superuser.com/questions/1447294) or migrate to MariaDB.

**Reply by Alien0-7** [reply]
I'm already using mysql_native_password
If you see the pastebin it also writes it

**Reply by TuxCoding** [reply]
Sorry, I thought it was the other way around. Could you post the plugin version, because the driver should support that authentication scheme.

**Reply by TuxCoding** [reply, marked as answer]
Okay, I guess I found the error now. Could you test the newer version once built.

**Comment by Alien0-7** [comment]
I have a question to ask you... but if I have 2 servers, one hub where players are registered and the other where players are not registered I must put it in both servers? or just the hub and the proxy?

**Reply by TuxCoding** [reply]
Depends on where the login commands should be forwarded. Usually on all server where your auth plugin is installed including the registration server.

**Reply by Alien0-7** [reply]
so if i have 2 server, Hub and Towny i have to put it only in the Hub server? or even in the proxy?

**Reply by TuxCoding** [reply]
As the proxy setup guide: Always on the proxy. Then on spigot servers where the auth plugin is.

**Reply by Alien0-7** [reply]
Ok, thank you so much!
